js/blocks.js

```diff
@@ -7051,6 +7050,9 @@ function Blocks(activity) {
             // console.debug('putting ' + this.blockList[blk].name + ' in the trash');
             this.blockList[blk].trash = true;
             this.blockList[blk].hide();
+            var title = this.blockList[blk].protoblock
```

Comment:
Does this static label always match the widget window title? Did you test?

Reply:
yes, it does. I have already tested this.

Reply:
@Ishakikani9117  Please check it for rhythm maker. 
If we send rhythm maker to trash , then widget window does not open for other rhythm maker block. Can you test this?

Reply:
@walterbender Also check for timbre , temperament widget. If once I move them to trash , the window not open for their other block .
One more issue : 
Is this intentional if I have two open widget window and I move one of their block to trash but both widget window get closed .

Reply:
It is a bug in how the window is hidden. style.display is never reset when the new widget tries to open. The bug was introduced by an earlier patch.

Reply:
> @walterbender Also check for timbre , temperament widget. If once I move them to trash , the window not open for their other block .
> One more issue :
> Is this intentional if I have two open widget window and I move one of their block to trash but both widget window get closed .

@b18050  I tried your solution, the phrase maker and custom mode widget are not being closed even we sent it to trash. and for other blocks widget disappears when you sent blocks to trash ,but when you open widgets for any other blocks ,the previous widget also 
shows up.

Reply:
@walterbender  Is it fixed now?

> It is a bug in how the window is hidden. style.display is never reset when the new widget tries to open. The bug was introduced by an earlier patch.

Reply:
Pushed a fix. Please test. (Should fix the same regression introduced by the new project button too.)

Reply:
@walterbender The widget window is not closing when block is send to trash.

Reply:
@walterbender same issue as @Ishakikani9117 was telling . The widget window shows up automatically after 2 seconds or so. Probably , this is due to overwriting session data in activity.js